rama-http/src/headers/x_robots_tag/robots_tag.rs

```diff
@@ -0,0 +1,102 @@
+use crate::headers::util::value_string::HeaderValueString;
+use crate::headers::x_robots_tag::custom_rule::CustomRule;
+use crate::headers::x_robots_tag::max_image_preview_setting::MaxImagePreviewSetting;
+use crate::headers::x_robots_tag::robots_tag_builder::RobotsTagBuilder;
+
+macro_rules! getter_setter {
+    ($field:ident, $type:ty) => {
+        paste::paste! {
+            pub fn [<$field>](&self) -> $type {
+                self.[<$field>]
+            }
+
```

Comment:
can you also provide the `(mut self, ...) -> Self` variants? as an additional way to set the vaues